File: 코딩 기초 트레이닝/배열 만들기4-p181918/배열 만들기4.py
# https://school.programmers.co.kr/learn/courses/30/lessons/181918

def solution(arr):
    stk = []

    i = 0

    while i < len(arr):
        if len(stk) == 0:
            stk.append(arr[i])
            i += 1
        else:
            if stk[len(stk)-1] < arr[i]:
                stk.append(arr[i])
                i += 1
            elif stk[len(stk)-1] >= arr[i]:
                stk.pop()

    return stk

File: 코딩 기초 트레이닝/배열 만들기4-p181918/배열 만들기4.js
// https://school.programmers.co.kr/learn/courses/30/lessons/181918

function solution(arr) {
    var stk = [];

    let i = 0;

    while(i < arr.length) {
        if(stk.length === 0) {
            stk.push(arr[i]);
            i += 1;
        } else {
            if(stk[stk.length-1] < arr[i]) {
                stk.push(arr[i]);
                i += 1;
            } else if(stk[stk.length-1] >= arr[i]) {
                stk.pop();
            }
        }
    }

    return stk;
}
